fecgov/openFEC
# Add filing_form and is_notice filters in sched_e.ScheduleEEfileView

## Summary (required)
- Resolves #[[_4140_](https://github.com/fecgov/openFEC/issues/4140)]
Add filter by filing_form and is_notice for  /schedules/schedule_e/efile endpoint.

_Include a summary of proposed changes._
1)modify public.real_efile_se_f57_vw
     LEFT JOIN real_efile.reps to get form type
2)Add column: filing_form
```
CASE
WHEN rep.form like 'F3X%' THEN 'F3X'::text
WHEN rep.form like 'F5%' THEN 'F5'::text
WHEN rep.form like 'F24%' THEN 'F24'::text
ELSE rep.form
END AS filing_form, 
```
3)Add column: is_notice
```
CASE
WHEN rep.form like 'F24%'
OR (rep.form like 'F5%' AND rep.rptcode is null) THEN true
ELSE false
END AS is_notice
```
## How to test the changes locally
1)checkout out branch and point to dev db
2)modify line 660 in itemized.py (point to temp view)
__tablename__ = 'real_efile_se_f57_vw_jl'

3)pytest
4)./manage.py runserver
5)check local endpoint: /schedules/schedule_e/efile/   under tag:  **independent expenditures**
make sure all filters work as expected. especially two new filters:filing_form and is_notice

*** Or you can use the following urls to test locally:
1)total count~6809 (this number maybe changed)
http://127.0.0.1:5000/v1/schedules/schedule_e/efile/?sort=-expenditure_date


2)filter by is_notice
a) is_notice=true, count~ 2523 
http://127.0.0.1:5000/v1/schedules/schedule_e/efile/?sort=-expenditure_date&is_notice=true

b)is_notice=false, count~4286
http://127.0.0.1:5000/v1/schedules/schedule_e/efile/?sort=-expenditure_date&is_notice=false

2523 +4286=6809


3)filter by filing_form 
a)filing_form=F3X, count~ 3013
http://127.0.0.1:5000/v1/schedules/schedule_e/efile/?sort=-expenditure_date&filing_form=F3X


b)filing_form=F5, count~ 1755
http://127.0.0.1:5000/v1/schedules/schedule_e/efile/?sort=-expenditure_date&filing_form=F5


c)filing_form=F5, is_notice=false, count~1273
http://127.0.0.1:5000/v1/schedules/schedule_e/efile/?sort=-expenditure_date&filing_form=F5&is_notice=false




d)filing_form=F5, is_notice=true, count~482
http://127.0.0.1:5000/v1/schedules/schedule_e/efile/?sort=-expenditure_date&filing_form=F5&is_notice=true



## Impacted areas of the application
endpoint: /schedules/schedule_e/efile/   under tag: **independent expenditures**
